src/lib/core/a11y/focus-trap.ts

```diff
@@ -114,4 +158,55 @@ export class FocusTrap {
 
     return null;
   }
+
+  private _createAnchor(): HTMLElement {
+    let anchor = document.createElement('div');
+    anchor.tabIndex = 0;
+    anchor.classList.add('cdk-visually-hidden');
+    anchor.classList.add('cdk-focus-trap-anchor');
+    return anchor;
+  }
+}
+
+
+/**
+ * Service that allows easy instantiation of focus traps.
+ */
+@Injectable()
+export class FocusTrapService {
+  constructor(private _checker: InteractivityChecker, private _ngZone: NgZone) { }
+
+  attach(element: HTMLElement, deferAnchors = false): FocusTrap {
+    return new FocusTrap(element, this._checker, this._ngZone, deferAnchors);
+  }
+}
+
+
+/**
+ * Directive for trapping focus within a region.
+ */
+@Directive({
+  selector: 'cdk-focus-trap, [cdkFocusTrap]',
```

Comment:
Can you split these into 2 separate directives that work like this:

`<cdk-focus-trap [disabled]="!isEnabled"></cdk-focus-trap>`
`<div [cdkTrapFocus]="isEnabled"></div>`

Reason being that we can't be guaranteed that the attribute won't be attached to some element that already assigns some other meaning to the `disabled` attribute. (also mark the element version deprecated and update any documentation/examples that refer to the element version).